app/Platform/Actions/MigrateAchievementIdsToDifferentGameId.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Actions;
+
+use App\Community\Enums\ArticleType;
+use App\Models\Achievement;
+use App\Models\Game;
+use App\Models\PlayerAchievement;
+use App\Models\PlayerGame;
+use App\Models\User;
+use App\Platform\Jobs\UpdatePlayerGameMetricsJob;
+use Illuminate\Support\Str;
+
+class MigrateAchievementIdsToDifferentGameId
+{
+    public function execute(array $achievementIds, int $gameId, User $user): void
+    {
+        // Determine which game(s) the achievements are coming from.
+        $oldGameIds = Achievement::whereIn('ID', $achievementIds)->select(['GameID'])->distinct()->pluck('GameID');
+
+        // Associate the achievements to the new game.
+        Achievement::whereIn('ID', $achievementIds)->get()->each(function ($achievement) use ($gameId) {
+            $achievement->update(['GameID' => $gameId]);
+        });
+
+        // Add an audit comment to the new game.
+        addArticleComment(
+            'Server',
+            ArticleType::GameModification,
+            $gameId,
+            "{$user->display_name} migrated " . Str::plural('achievement', count($achievementIds)) . ' ' .
+                implode(',', $achievementIds) . ' from ' .
+                Str::plural('game', count($oldGameIds)) . ' ' . $oldGameIds->implode(',') . '.'
+        );
+
+        // Update all affected player game metrics.
+        $affectedUserIds = PlayerAchievement::whereIn('achievement_id', $achievementIds)
+            ->select(['user_id'])
+            ->distinct()
+            ->pluck('user_id');
+        foreach ($affectedUserIds as $userId) {
+            // Ensure player_game entries exist for the new game for all affected users.
+            if (!PlayerGame::where('game_id', $gameId)->where('user_id', $userId)->exists()) {
+                $playerGame = new PlayerGame(['user_id' => $userId, 'game_id' => $gameId]);
+                $playerGame->save();
+            }
+
+            dispatch(new UpdatePlayerGameMetricsJob($userId, $gameId));
+            foreach ($oldGameIds as $oldGameId) {
+                dispatch(new UpdatePlayerGameMetricsJob($userId, $oldGameId));
+            }
+        }
+
+        // Update achievement_sets associated with the given achievement IDs.
+        $upsertAction = new UpsertGameCoreAchievementSetFromLegacyFlags();
+        $upsertAction->execute(Game::find($gameId));
```

Comment:
I assume it's fine, but I noticed that moving achievements to a set with no achievements generates an audit log entry for the new game, which is displayed awkwardly:
![image](https://github.com/user-attachments/assets/7177cfd6-8169-4c30-a910-13aa1dbbed60)

Reply:
I didn't see that earlier -- good spot. 

It's probably safe to ignore for now. This appears to be a side effect of how we're handling double writes. We'll likely reach a state in the future where this upsert action is the default behavior, and `Flags` is dropped from achievements altogether (tracked instead via achievement sets).

@luchaos put this TODO in the `Achievement` model:
```
// TODO drop Flags, derived from being included in an achievement set
```
It's worth noting that `achievement_set_achievements` has no notion of whether the achievement is published or not. We'll likely need to store that info in `achievement_set_achievements` rather than `Achievements`, as an achievement should probably never be orphaned from a set.